<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="It is an innovative tool designed to facilitate quality control for spatialomics technology data.">
<title>QC metrics and support for In-Situ Spatial data across different platforms • SpatialQC</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="QC metrics and support for In-Situ Spatial data across different platforms">
<meta property="og:description" content="It is an innovative tool designed to facilitate quality control for spatialomics technology data.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">SpatialQC</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/SpatialTouchstone-tutorial.html">Introduction to SpatialQC</a>
    <a class="dropdown-item" href="articles/SpatialTouchstone.html">SpatialQC</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Margery0011/SpatiaTouchstone/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="spatialqc">SpatialQC<a class="anchor" aria-label="anchor" href="#spatialqc"></a>
</h1></div>
<p><code>SpatialQC</code>, is a package that supports loading Spatial In-Situ datasets and calculating Quality Control metrics to aid understanding of data quality.</p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Spatial omics technologies are revolutionizing how we understand biological tissues, enabling the mapping of molecular compositions at unprecedented spatial resolutions. However, with great power comes great complexity. The interpretation of spatial omics data requires robust quality control measures to ensure the reliability and reproducibility of the measurements. <code>SpatialQC</code> is developed to meet this need, providing a comprehensive suite of quality control metrics specifically tailored for In-situ spatial omics data (e.g CosMx, Xenium).</p>
</div>
<div class="section level2">
<h2 id="how-to-install">How to install<a class="anchor" aria-label="anchor" href="#how-to-install"></a>
</h2>
<p><code>remotes::install_github('https://github.com/Center-for-Spatial-OMICs/SpatialQC')</code></p>
<div class="section level3">
<h3 id="for-dedenpency-packages">for dedenpency packages<a class="anchor" aria-label="anchor" href="#for-dedenpency-packages"></a>
</h3>
<p>For users wanting to use annotation functions in <code>SpatialQC</code>, Package <code>RCTD</code> &amp; <code>InSituType</code> are recommended.</p>
<p>To install:</p>
<pre><code><span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"dmcable/RCTD"</span><span class="op">)</span></span></code></pre>
<pre><code><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"https://github.com/Nanostring-Biostats/InSituType"</span><span class="op">)</span></span></code></pre>
<p><em>Notes:</em></p>
<p>Package <code>InSituType</code> needs a C++ compiler which can be installed as follows:</p>
<div class="section level4">
<h4 id="gcc-and-gfortran-are-required-for-mac-users-to-install-insitutype">gcc and gfortran are required for Mac users to install <code>InSituType</code>
<a class="anchor" aria-label="anchor" href="#gcc-and-gfortran-are-required-for-mac-users-to-install-insitutype"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">/bin/bash</span> <span class="at">-c</span> <span class="st">"</span><span class="va">$(</span><span class="ex">curl</span> <span class="at">-fsSL</span> https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh<span class="va">)</span><span class="st">"</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">brew</span> install gcc</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">gcc</span> <span class="at">--version</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/.R</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">touch</span> ~/.R/Makevars</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="fu">vi</span> ~/.R/Makevars</span></code></pre></div>
<p>Add the following lines to your Makevars file:</p>
<pre><code>CC=/opt/homebrew/bin/gcc-14
CXX=/opt/homebrew/bin/g++-14
CXX11=/opt/homebrew/bin/g++-14
CXX14=/opt/homebrew/bin/g++-14
CXX17=/opt/homebrew/bin/g++-14
FC=/opt/homebrew/bin/gfortran-14
F77=/opt/homebrew/bin/gfortran-14</code></pre>
<p>For others, please use our <a href="https://github.com/Center-for-Spatial-OMICs/SpatialQC/blob/main/R/DependencyPackages.R" class="external-link">Rscipts</a></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="functions">Functions<a class="anchor" aria-label="anchor" href="#functions"></a>
</h2>
<div class="section level3">
<h3 id="readspatial">readSpatial<a class="anchor" aria-label="anchor" href="#readspatial"></a>
</h3>
<p>Reads and preprocesses spatial transcriptomics data from a specified path for a given imaging based platform. The function supports ‘Xenium’ and ‘CosMx’ platforms, performing platform-specific loading and preprocessing steps. This includes loading the data, annotating it with sample metadata and processing cell metadata. For ‘Xenium’, it adds additional cell metadata and tissue coordinates as an embedding for custom plotting. For ‘CosMx’, it separates targeting and non-targeting probes, adds additional cell metadata and includes tissue coordinates as an embedding. ‘Merscope’ platform support is under development. Regardless of platform, data is stored in an assay named “RNA” for convenient and for each platform, this table will be used by subsequent functions.</p>
</div>
<div class="section level3">
<h3 id="readtxmeta">readTxMeta<a class="anchor" aria-label="anchor" href="#readtxmeta"></a>
</h3>
<p>Reads transcriptome metadata from a specified path, depending on the platform specified. Currently supports ‘Xenium’ and ‘CosMx’ platforms. For ‘Xenium’, it reads ‘transcripts.csv.gz’ and renames the ‘feature_name’ column to ‘target’ for consistency. For ‘CosMx’, it reads the appropriate transcriptome file matched by ‘*tx_file.csv.gz’. ‘Merscope’ platform support is under development. For each platform, this table will be used by subsequent functions.</p>
</div>
<div class="section level3">
<h3 id="getglobalfdr">getGlobalFDR<a class="anchor" aria-label="anchor" href="#getglobalfdr"></a>
</h3>
<p>It calculate ‘specificity’ as Global False Discovery Rate (FDR) for specified features (or all features by default) in a Seurat object. It leverages transcript localization data to calculate FDR based on the proportion of negative control or blank barcode expressions compared to the expression of actual genes.</p>
</div>
<div class="section level3">
<h3 id="gettxpercell">getTxPerCell<a class="anchor" aria-label="anchor" href="#gettxpercell"></a>
</h3>
<p>This function calculates the mean number of transcripts per cell for a specified set of features (genes) in a Seurat object. If no features are specified, the function defaults to using all targets available within the RNA assay of the provided Seurat object. It’s a useful metric for assessing the overall transcriptional activity within the sampled cells.</p>
</div>
<div class="section level3">
<h3 id="gettxperarea">getTxPerArea<a class="anchor" aria-label="anchor" href="#gettxperarea"></a>
</h3>
<p>Calculates the mean number of transcripts per unit area for specified features (genes) in a Seurat object. This calculation provides insight into the density of transcriptional activity relative to cell area, offering a normalized measure of gene expression that accounts for differences in cell size.</p>
</div>
<div class="section level3">
<h3 id="gettxpernuc">getTxPerNuc<a class="anchor" aria-label="anchor" href="#gettxpernuc"></a>
</h3>
<p>This function calculates the mean number of transcripts localized within the nucleus for a specified set of features (genes) across all cells in a given dataset. This measurement is specific to spatial transcriptomics data where the distinction between nuclear and cytoplasmic transcript localization can be made.</p>
</div>
<div class="section level3">
<h3 id="getmeanexpression">getMeanExpression<a class="anchor" aria-label="anchor" href="#getmeanexpression"></a>
</h3>
<p>Computes the mean expression levels for a specified list of features (genes) and control probes within a Seurat object. Computes the mean expression levels for a specified list of features (genes) and control probes within a Seurat object. This function separately calculates the mean expression for both target genes and control probes, then combines these results into a single data frame.</p>
</div>
<div class="section level3">
<h3 id="getmeansignalratio">getMeanSignalRatio<a class="anchor" aria-label="anchor" href="#getmeansignalratio"></a>
</h3>
<p>Computes the log-ratio of the mean expression levels of specified genes (or all genes if none are specified) to the mean expression levels of negative control probes within a Seurat object. This metric can provide insights into the overall signal strength relative to background noise levels.</p>
</div>
<div class="section level3">
<h3 id="getcelltxfraction">getCellTxFraction<a class="anchor" aria-label="anchor" href="#getcelltxfraction"></a>
</h3>
<p>Calculates the fraction of transcripts in cells for a given set of features or all features. The function works by reading transcript metadata for the given sample and platform, then filtering for the specified features (if any). For each platform, it identifies transcripts that are not assigned to any cell (‘UNASSIGNED’ for Xenium, ‘None’ for CosMx, etc.) and calculates the fraction of transcripts that are assigned to cells.</p>
</div>
<div class="section level3">
<h3 id="getmaxratio">getMaxRatio<a class="anchor" aria-label="anchor" href="#getmaxratio"></a>
</h3>
<p>Identifies the maximum mean expression value among the specified features (or all features if none are specified) and calculates its log-ratio to the mean expression value of negative control probes. This log-ratio reflects the dynamic range of the dataset, indicating the spread between the</p>
</div>
<div class="section level3">
<h3 id="getmaxdetection">getMaxDetection<a class="anchor" aria-label="anchor" href="#getmaxdetection"></a>
</h3>
<p>Calculates the distribution of maximal values by identifying the maximal expression values across the specified set of features (or all features if none are specified) within the dataset. This function illustrates the upper bounds of gene expression, providing crucial information for assessing the dataset’s dynamic range and the sensitivity of detection methods used in the experiment. The function aggregates these maximal values and presents them in a data frame, facilitating further analysis of the expression distribution and detection efficiency across different genes.</p>
</div>
<div class="section level3">
<h3 id="getmecr">getMECR<a class="anchor" aria-label="anchor" href="#getmecr"></a>
</h3>
<p>Calculates the Mutually Exclusive Co-expression Rate (MECR) Implementation. This function is based on a predefined set of markers and their associated cell types, computing the rate at which pairs of markers are expressed in mutually exclusive patterns within cells. The calculation is limited to markers found in the dataset and can be adjusted to focus on a subset by specifying it. The result is a single MECR value that quantifies the overall mutually exclusive co-expression pattern, rounded to three decimal places. This is particularly useful for understanding cellular heterogeneity and the specificity of marker gene expression within a spatial transcriptomics dataset, offering insights into the cellular composition and the distinctiveness of cell-type-specific gene expression patterns.</p>
</div>
<div class="section level3">
<h3 id="getmorans">getMorans<a class="anchor" aria-label="anchor" href="#getmorans"></a>
</h3>
<p>Calculates spatial autocorrelation using Moran’s I. This function processes gene-targeting probes by converting the input data into a <code>SingleCellExperiment</code> object, then into a <code>SpatialFeatureExperiment</code> object. Spatial coordinates are specified and used to further process the data, including normalization and nearest neighbor graph construction. Subsequently, Moran’s I is calculated for each feature using the <code>Voyager</code> package. This process is repeated for control probes. The results for both gene-targeting and control probes are combined into a single data frame. The spatial autocorrelation analysis provides Moran’s I values for each feature, indicating the degree of spatial clustering, and is performed separately for gene-targeting probes and control probes, allowing for a comparison of spatial patterns in gene expression and background noise.</p>
</div>
<div class="section level3">
<h3 id="getsilhouettewidth">getSilhouetteWidth<a class="anchor" aria-label="anchor" href="#getsilhouettewidth"></a>
</h3>
<p>Calculates silhouette width as cluster evaluation, providing a metric for assessing the quality of the clustering. By clustering the data and calculating the silhouette width, this function evaluates how similar an object is to its own cluster compared to other clusters. The silhouette width ranges from -1 to 1, where a high value indicates that the object is well matched to its own cluster and poorly matched to neighboring clusters. This measure is useful for determining the appropriate number of clusters in the data, guiding the interpretation of spatial patterns and cellular heterogeneity within the dataset.</p>
</div>
<div class="section level3">
<h3 id="getsparsity">getSparsity<a class="anchor" aria-label="anchor" href="#getsparsity"></a>
</h3>
<p>Shows the sparsity (as a count or proportion) of a matrix, which is a crucial aspect of high-dimensional data analysis, especially in single-cell and spatial transcriptomics. A high sparsity value (close to 1) means that the majority of the matrix elements are zero, indicating a large number of non-detected genes or features in the dataset. Understanding sparsity is important for choosing the appropriate data normalization and transformation techniques, which can significantly impact downstream analysis and interpretations. This function calculates the sparsity of the RNA counts matrix in a Seurat object, providing insights into the data’s density and informing decisions on data preprocessing steps.</p>
</div>
<div class="section level3">
<h3 id="getentropy">getEntropy<a class="anchor" aria-label="anchor" href="#getentropy"></a>
</h3>
<p>Calculates the entropy of the expression matrix in a Seurat object, offering a measure of the distribution’s uniformity across the detected genes. Entropy is a concept from information theory that measures the uncertainty or disorder within a dataset. In the context of transcriptomics, a higher entropy value can indicate a more heterogeneous expression profile across cells, reflecting the complexity and diversity of transcriptional states within the sample. By quantifying entropy, this function provides a summary statistic of the dataset’s overall expression diversity, aiding in the understanding of cellular complexity and the identification of transcriptionally distinct cell populations.</p>
</div>
<div class="section level3">
<h3 id="plot-function-in-utilsr">Plot function in utils.R<a class="anchor" aria-label="anchor" href="#plot-function-in-utilsr"></a>
</h3>
<ul>
<li>plotSampleLabel</li>
<li>plotPanelSize</li>
<li>plotCellCount</li>
<li>plotTxPerCell</li>
<li>plotTxPerArea</li>
<li>plotTxPerNuc</li>
<li>plotTxPerCellNorm</li>
<li>plotFractionTxInCell</li>
<li>plotSignalRatio</li>
<li>plotMeanExpression</li>
<li>plotMaxExpression</li>
<li>plotMECR</li>
<li>plotMorans</li>
<li>plotSilhouette</li>
<li>plotSparsity</li>
<li>plotEntropy</li>
</ul>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Margery0011/SpatiaTouchstone/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/Margery0011/SpatiaTouchstone/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SpatialQC</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Felipe Segato Dezem <br><small class="roles"> Author, maintainer </small>  </li>
<li>Yutian Liu <br><small class="roles"> Author </small>  </li>
<li>Maycon Marcao <br><small class="roles"> Author </small>  </li>
<li>David Cook <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Felipe Segato Dezem, Yutian Liu, Maycon Marcao, David Cook.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
